src/services/request-service.ts

```diff
@@ -0,0 +1,75 @@
+import Promise = require('bluebird');
+import r = require('rethinkdb');
+import uuid = require('uuid');
+import EmailService = require('./email-service');
+import nodemailer = require('nodemailer');
+import q = require('../dbutils/query');
+import models = require('../models/models');
+import errors = require('../errors/errors');
+import config = require('../config');
+
+
+module RequestService {
```

Comment:
So I refactored the code to use branches and did some testing.  Branches look they like they are still prone to race conditions. 

I ran the following code:
    for(var i = 0; i < 100; ++i){
      requestService.createRequest("123", "123")
        .then(console.log)
        .catch(Error, (err)=>{});
    }

The branches caught more of the conflicts than the previous method, but it still allowed 7 duplicates to make it through.